sdk/python/kfp/_client.py

```diff
@@ -139,6 +142,11 @@ def __init__(self, host=None, client_id=None, namespace='kubeflow', other_client
     self._experiment_api = kfp_server_api.api.experiment_service_api.ExperimentServiceApi(api_client)
     self._pipelines_api = kfp_server_api.api.pipeline_service_api.PipelineServiceApi(api_client)
     self._upload_api = kfp_server_api.api.PipelineUploadServiceApi(api_client)
+    if self._context_setting['namespace'] == "" and self.get_kfp_healthz().get('multi_user') is True and self._is_ipython() is True:
```

Comment:
@Bobgy The namespace is currently set by reading `/var/run/secrets/kubernetes.io/serviceaccount/namespace`. I added the check for ipython for situations where th KFP SDK is used outside the cluster in a multi-user environment as it would then not be able to get the current namespace. Another option would be to use the Kubernetes client to get the namespace.

Reply:
@DavidSpek
But the IPython check isn't doing what you describe. I could be running in an IPython kernel on my laptop.
Reading the `/var/run/secrets/kubernetes.io/serviceaccount/namespace` file is the key. You cannot set the namespace unless reading this file succeeds.

Reply:
@elikatsis Your right in that it isn't the correct way to handle this. I think this can best be solved by checking if the `/var/run/secrets/kubernetes.io/serviceaccount/namespace` file exists instead.

Reply:
@DavidSpek, you could just `try/except` the reading of the file inside that `if`.
I.e., put the `with open(...)...` in a `try/except` block.

Reply:
@elikatsis That was also my first intuition but I wasn't sure how to handle the `except`. I now have:
```
except:
    print('Failed to automatically set namespace.')
    pass
```